src/Http/Response.php

```diff
@@ -1137,6 +1137,22 @@ public function type($contentType = null)
         return $contentType;
     }
 
+    /**
+     * Sets a content type into the map.
+     *
+     * E.g.: setType('xhtml' => ['application/xhtml+xml', 'application/xhtml', 'text/xhtml'])
+     *
+     * This is needed for RequestHandlerComponent and recognition of types.
+     *
+     * @param string $type Content type.
+     * @param string|array $definition Definition of the content type.
+     * @return void
+     */
+    public function setType($type, $definition)
```

Comment:
Why do we need `setType()` when there is `withType()` ?

Reply:
See the ticket referenced. The type() deprecation lost a way to set new type definitions.
As such RequestHandler would just not be able to serve those anymore without calling

    $this->response->setType('ical', 'my/ical');

etc

Reply:
Okay, in that case `addType()` would be a more intuitive name IMO.

Reply:
I don't think the fact that `addType()` can overwrite existing mapping will cause confusion to anymore. Someone requiring to overwrite existing type mapping is unlikely anyway. `mapType()` would have been the ideal name but method with that name already exists for another purpose :slightly_frowning_face:.

Reply:
How about the name `setTypeMap()` ?

Reply:
Modified, also added tests for it.